docs/book/src/plugins/external-plugins.md

```diff
@@ -0,0 +1,155 @@
+	Command string `json:"command"`
+
+	// Metadata contains the plugin specific help text that the plugin returns to Kubebuilder when it receives
+	// `--help` flag from Kubebuilder.
+	Metadata plugin.SubcommandMetadata `json:"metadata"`
+
+	// Universe in the PluginResponse represents the updated file contents that was written by the plugin.
+	Universe map[string]string `json:"universe"`
+
+	// Error is a boolean type that indicates whether there were any errors due to plugin failures.
+	Error bool `json:"error,omitempty"`
+
+	// ErrorMsgs contains the specific error messages of the plugin failures.
+	ErrorMsgs []string `json:"errorMsgs,omitempty"`
+
+	// Flags contains the plugin specific flags that the plugin returns to Kubebuilder when it receives
+	// a request for a list of supported flags from Kubebuilder
+	Flags []Flag `json:"flags,omitempty"`
+}
+```
+
+<aside class="note">
+<h1>Caution</h1>
+
+When writing your own external plugin, you **should not** directly echo or print anything to the stdout. 
+
+This is because Kubebuilder and your plugin are communicating with each other via `stdin` and `stdout` using structured `JSON` data.
+Any additional information sent to stdout (such as debug messages or logs) that's not part of the expected PluginResponse JSON structure may cause parsing errors when Kubebuilder tries to read and decode the response from your plugin.
+
+If you need to include logs or debug messages while developing your plugin, consider writing these messages to a log file instead.
+
+</aside>
+
+## How to use it?
+
+### Prerequisites
+
+- You have built and installed your Kubebuilder binary.
+
+- You have built and installed your external plugin as an **executable / binary**.
+
+- Your plugin is put in a path following the GVK schema:
+```sh
+/path1/path2/${name}/${version} 
+```
+
+### Usage:
+
+The external plugin supports the same subcommands as kubebuilder already provides:
+- `init`: project initialization.
+- `create api`: scaffold Kubernetes API definitions.
+- `create webhook`: scaffold Kubernetes webhooks.
+
+
+Also, if the plugin is put in a configured path, you need to use a env variable `$EXTERNAL_PLUGINS_PATH` to tell Kubebuilder where to search for the plugin binary.
+
+Otherwise, Kubebuilder would search for the plugins in a default path based on your OS:
+1. On Linux:
+```sh
+$XDG_CONFIG_HOME/kubebuilder/plugins/${name}/${version}
+```
+
+2. On OSX
+```sh
+~/Library/Application Support/kubebuilder/plugins/${name}/${version}
+```
+
+**Examples**
+```sh
+# Initialize a new project with the external plugin named `sampleplugin`
+EXTERNAL_PLUGINS_PATH=/path1/path2/sampleplugin/v1/sampleplugin kubebuilder init --plugins sampleplugin/v1
+
+# Create a new API with the above external plugin with a customized flag `number`
+EXTERNAL_PLUGINS_PATH=/path1/path2/sampleplugin/v1/sampleplugin kubebuilder create api --plugins sampleplugin/v1 --number 2
```

Comment:
Instead of `/path1/path2` can we specify something else as it may cause confusion. For example:
```suggestion
EXTERNAL_PLUGINS_PATH=<home-dir>/sampleplugin/v1/sampleplugin kubebuilder create api --plugins sampleplugin/v1 --number 2
```

This comment applies to all other places where `/path1/path2` is mentioned.